packages/pwa-kit-dev/bin/pwa-kit-dev.js

```diff
@@ -284,13 +277,63 @@ const main = () => {
     program
         .command('test')
         .description('test the project')
-        .action((_, {args}) => {
+        .action((_, opts) => {
+            const {args} = opts
             const jest = p.join(require.resolve('jest'), '..', '..', '..', '.bin', 'jest')
             execSync(
                 `${jest} --passWithNoTests --maxWorkers=2${args.length ? ' ' + args.join(' ') : ''}`
             )
         })
 
+    program
+        .command('logs')
+        .description(`tail environment logs`)
+        // TODO: add a --tail flag and make -p optional. get the default from package.json
+        .option('-p, --project <project_slug>', 'the project slug', (val) => {
+            if (!(typeof val === 'string') && val.length > 0) {
+                throw new program.InvalidArgumentError(`"${val}" cannot be empty`)
+            } else {
+                return val
+            }
+        })
+        .requiredOption('-e, --environment <environment_slug>', 'the environment slug', (val) => {
+            if (!(typeof val === 'string') && val.length > 0) {
+                throw new program.InvalidArgumentError(`"${val}" cannot be empty`)
+            } else {
+                return val
+            }
+        })
+        .action(async (_, opts) => {
+            const {project, environment, cloudApiBase, credentialsFile} = opts.optsWithGlobals()
```

Comment:
Really nice – I like that `optsWithGlobals` makes the order of the args insignificant. Couldn't work out how to do this myself the other day.

There are a set of commands in here for which these "global" flags can never be of interest – basically anything that is not an "MRT Command" (build, lint, etc).

I think that's harmless, but I came up with this `mrtCommand` helper [here](https://github.com/SalesforceCommerceCloud/pwa-kit/pull/763/files#diff-52cf52d1c1e84c76145d9b0a69946921f732dbff0fa9a1295946093b0fe2779cR93) that might be worth thinking about.

The only advantage I can think of with the `mrtCommand` approach is that the `--cloudURLBase` still appear in the command-specific help. Here's what we see with your approach now:

```
(venv) ➜  pwa-kit-dev git:(tail-logs) ./bin/pwa-kit-dev.js logs --help
Usage: pwa-kit-dev logs [options]

tail environment logs

Options:
  -p, --project <project_slug>          the project slug
  -e, --environment <environment_slug>  the environment slug
  -h, --help                            display help for command
```

It's a nice-to-have.

Reply:
Interesting, I like the `mrtCommand` idea. I think your point about the global options now appearing for command-specific help is important, cause I also made `credentialsFile` global, which means I've taken it out of the `push` and `save-credentials` help. And the wrapper seems similar to [what other people are doing](https://github.com/tj/commander.js/issues/476#issuecomment-992048035) to get around that problem.

Reply:
I also like the fancy color-coded logs you have there! I wanna give that a try too :)